daft/filesystem.py

```diff
@@ -6,17 +6,20 @@
 import pathlib
 import sys
 import urllib.parse
-from typing import Any, Literal
+from typing import TYPE_CHECKING, Any, Literal
 
-import fsspec
-from fsspec.registry import get_filesystem_class
 from pyarrow.fs import FileSystem, LocalFileSystem, S3FileSystem
 from pyarrow.fs import _resolve_filesystem_and_path as pafs_resolve_filesystem_and_path
 
 from daft.daft import FileFormat, FileInfos, IOConfig, io_glob
+from daft.lazy_import import LazyImport
 from daft.table import MicroPartition
 
+if TYPE_CHECKING:
+    import fsspec
+
 logger = logging.getLogger(__name__)
+fsspec = LazyImport("fsspec")
```

Comment:
As a follow on, we should look into if we can just depreciate fsspec. I'm not sure if we need it as a default dep anymore 

Any thoughts @jaychia?